npm/test/index.test.ts

```diff
@@ -1,4 +1,76 @@
-// Replace this with a real test when we need it
-test("a sample test", () => {
-  expect(1 + 2).toBe(3);
-});
+import { ClientFileMetadataUpload } from "../src/clientfilemetadataupload"
+
+const mockAuth = {
+  getToken: jest.fn()
+}
+
+import { KeycloakInstance } from "keycloak-js"
+import { renderModules } from "../src/index"
+import SpyInstance = jest.SpyInstance
+import { UploadFiles } from "../src/upload"
+import { GraphqlClient } from "../src/graphql"
+
+jest.mock("../src/auth", () => mockAuth)
+
+beforeEach(() => jest.resetModules())
+
+const mockKeycloak: KeycloakInstance<"native"> = {
+  init: jest.fn(),
+  login: jest.fn(),
+  logout: jest.fn(),
+  register: jest.fn(),
+  accountManagement: jest.fn(),
+  createLoginUrl: jest.fn(),
+  createLogoutUrl: jest.fn(),
+  createRegisterUrl: jest.fn(),
+  createAccountUrl: jest.fn(),
+  isTokenExpired: jest.fn(),
+  updateToken: jest.fn(),
+  clearToken: jest.fn(),
+  hasRealmRole: jest.fn(),
+  hasResourceRole: jest.fn(),
+  loadUserInfo: jest.fn(),
+  loadUserProfile: jest.fn(),
+  token: "fake-auth-token"
+}
+
+beforeEach(() => {
+  jest.resetModules()
+})
+
+test("renderModules calls authorisation when upload form present on page", async () => {
+  const spyAuth = jest
+    .spyOn(mockAuth, "getToken")
+    .mockImplementation(() => Promise.resolve(mockKeycloak))
+
```

Comment:
You can get rid of the spy here if you `import { getToken } from "../src/auth"` and then:
```
const token = getToken as jest.Mock
token.mockImplementation(() => Promise.resolve(mockKeycloak))

```

Reply:
resolved